superset/utils/machine_auth.py

```diff
@@ -43,15 +43,15 @@
 class MachineAuthProvider:
     def __init__(
         self,
-        auth_webdriver_func_override: Callable[[WebDriver, User], WebDriver],
-        auth_context_func_override: Callable[[BrowserContext, User], BrowserContext],
+        auth_webdriver_func_override: Callable[
+            [WebDriver | BrowserContext, User], WebDriver | BrowserContext
+        ],
```

Comment:
The way we were using these was IMO incorrect - previously `auth_webdriver_func_override` was in fact optional, not always a `Callable`. So I propose changing the sig to make these optional, and defaulting them to `None`, like so:

```python
auth_webdriver_func_override: Callable[[WebDriver, User], WebDriver] | None = None,
```